src/PointwiseFunctions/GeneralRelativity/Psi4.cpp

```diff
@@ -69,27 +71,62 @@ void psi_4(const gsl::not_null<Scalar<ComplexDataVector>*> psi_4_result,
       inverse_spatial_metric, cov_deriv_extrinsic_curvature, r_hat,
       inverse_projection_tensor, projection_tensor, projection_up_lo, 1.0);
 
-  auto& x_coord = get<::Tags::TempI<0, 3, Frame>>(temp_buffer);
+  // Gram-Schmidt x_hat
+  auto& x_coord = get<::Tags::TempI<1, 3, Frame>>(temp_buffer);
   x_coord.get(0) = 1.0;
   x_coord.get(1) = x_coord.get(2) = 0.0;
-  auto& magnitude_x = get<::Tags::TempScalar<0>>(temp_buffer);
-  magnitude(make_not_null(&magnitude_x), x_coord, spatial_metric);
-  auto& x_hat = get<::Tags::TempI<0, 3, Frame>>(temp_buffer);
-  tenex::evaluate<ti::I>(make_not_null(&x_hat), x_coord(ti::I) / magnitude_x());
+  auto& x_component = get<::Tags::TempScalar<1>>(temp_buffer);
+  dot_product(make_not_null(&x_component), x_coord, r_hat, spatial_metric);
+  auto& x_hat = get<::Tags::TempI<2, 3, Frame>>(temp_buffer);
+  tenex::evaluate<ti::I>(make_not_null(&x_hat),
+                         x_coord(ti::I) - (x_component() * r_hat(ti::I)));
+  auto& magnitude_x = get<::Tags::TempScalar<1>>(temp_buffer);
+  magnitude(make_not_null(&magnitude_x), x_hat, spatial_metric);
+  for (size_t j = 0; j < 3; j++) {
+    for (size_t i = 0; i < get(magnitude_x).size(); i++) {
+      if (magnitude_x.get()[i] != 0.0) {
+        x_hat.get(j)[i] = x_hat.get(j)[i] / magnitude_x.get()[i];
+      } else {
+        x_hat.get(j)[i] = 0.0;
+      }
+    }
+  }
 
   Variables<tmpl::list<::Tags::TempI<0, 3, Frame, ComplexDataVector>,
                        ::Tags::TempI<1, 3, Frame, ComplexDataVector>>>
       y_hat_buffer{get<0>(inertial_coords).size()};
+
+  // Grad-Schmidt y_hat
```

Comment:
Clarify comment: state something like `Gram-Schmidt y_hat, i.e. make y_hat a unit vector orthogonal to r_hat and x_hat`